code-challenges.js

```diff
@@ -13,21 +26,53 @@ var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+let numsOnly = (value) => {
+    if(typeof(value)=== 'number' && value % 2 !== 0){
+        return value
+    }
+}
+// .filter() here that will create an array with only numbers and then put them in order from least to greatest with .sort()
+let oddNumsOnly = (array) => {
+    return array.filter(numsOnly).sort(function(a, b){return a-b})
+}
 
-
+console.log(oddNumsOnly(fullArr1))
+console.log(oddNumsOnly(fullArr2))
 
 // --------------------3) Create a function that takes in a string of a single word and returns the middle letter of the word. If the word is an even number of letters, return the two middle letters.
 
 var middleLetters1 = "hello"
 // Expected output: “l”
 var middleLetters2 = "rhinoceros"
 // Expected output: “oc”
+const centerLetter = (string) => {
+        let oddCoreLetter = string.length / 2
+        let evenCoreLetter = string.length / 2 - 1
+         if (string.length % 2 !==0) {
+            return string.charAt(oddCoreLetter)
+        } else {
+            return string.charAt(evenCoreLetter) + string.charAt(oddCoreLetter)
 
-
+        }
+}
+console.log(centerLetter(middleLetters1))
+console.log(centerLetter(middleLetters2))
 
 
 // --------------------4) Create a CLASS to get the area of a sphere. Create three spheres with different radi as test cases. Area of a sphere =  4πr^2 (four pi r squared)
+// class Spheres {
+//     constructor(){
+//         this.sphere1 = 12
+//         this.sphere2 = 668
+//         this.sphere2 = 1024
 
+//     formula1 (){
+//     return 4 * 3.14 * (this.sphere1 * sphere1)
+//     }
+
+
+//     }
+//     }
```

Comment:
I see where you're going with this! :D I would encourage you to try and finish this up! 
Thanks for letting us know in the description of the PR that this was unfinished.